### ISSUE TITLE
Media Modal: Keyboard navigation gets stuck on search icon

### ISSUE THREAD

**Issue by ryelle** [Automattic/wp-calypso#26280]
#### Steps to reproduce
1. Create or edit a post, `/page/:site`
2. Using only a keyboard, try to add an image – hit <kbd>ctrl</kbd> + <kbd>opt</kbd> + <kbd>m</kbd> to open media modal
3. Tab through the header (media types & search) to get into the main section, aiming to land on and select an image
4. Can't tab past the search icon

#### What I expected

I should be able to tab past the search icon so that I can select "Add New" or existing images to add to my post.

#### What happened instead

I'm stuck on Search, and even after tabbing forward multiple times, one tab back puts me back on the media types (so I know I'm not tabbing into invisible elements). Additionally, I can open the search bar, but am still stuck -- I can't tab forward into the search field.

The menu + search pattern _does_ work on other pages, for example the page list view  `/pages/:site` - I can tab through that menu, past the search icon, and into the list of pages.

#### Browser / OS version

Safari 11.1.2 – strangely enough this does not happen in Chrome, or if I use Safari with VoiceOver.

#### Screenshot / Video

You can watch my focus move with the keys I press in the corner, the arrow keys being tab forward & tab back:

https://cloudup.com/cP-PCCkTQHs
